rustfmt-core/rustfmt-lib/src/formatting/syntux/parser.rs

```diff
@@ -177,7 +177,9 @@ impl<'a> Parser<'a> {
                 Ok(_attrs) => (),
                 Err(mut e) => {
                     e.cancel();
-                    sess.reset_errors();
+                    if sess.can_reset_errors() {
+                        sess.reset_errors();
+                    }
```

Comment:
I'm not sure if this (and the similar change in `parse_mod_items` below) could cause any unintended side effects.

The only use case I could think of would be an instance where rustfmt is seeing, and deliberately swallowing, a parser error that the rustc parser was unable to recover from. Since this parsing logic is running on all submods now that the rustc parser doesn't parse out-of-line mods, I went ahead and changed this as I can't think of any reason why we _would_ want to ignore such parsing errors.